utils/formatters/number.ts

```diff
@@ -24,47 +19,49 @@ export type FormatCurrencyOptions = {
 	minDecimals?: number;
 	maxDecimals?: number;
 	sign?: string;
-	currencyKey?: CurrencyKey;
+	currencyKey?: string;
 };
 
 const DEFAULT_CURRENCY_DECIMALS = 2;
 export const SHORT_CRYPTO_CURRENCY_DECIMALS = 4;
 export const LONG_CRYPTO_CURRENCY_DECIMALS = 8;
 
-export const getDecimalPlaces = (value: NumericValue) =>
-	(value.toString().split('.')[1] || '').length;
-
-export const toBigNumber = (value: NumericValue) => new BigNumber(value);
+export const getDecimalPlaces = (value: WeiSource) => (value.toString().split('.')[1] || '').length;
 
-export const zeroBN = toBigNumber(0);
+export const zeroBN = wei(0);
 
 // TODO: implement max decimals
-export const formatNumber = (value: NumericValue, options?: FormatNumberOptions) => {
+export const formatNumber = (value: WeiSource, options?: FormatNumberOptions) => {
 	const prefix = options?.prefix;
 	const suffix = options?.suffix;
 
+	let weiValue = wei(0);
+	try {
+		weiValue = wei(value);
+	} catch {}
```

Comment:
empty catch?

Reply:
yea because wei throws an exception if it gets a non-numeric string

Reply:
do we want to `console.error("Non-numeric string")` ?

Reply:
probably shouldn't because possibility of entry of non-numeric strings is a normal part of string data entry.

it would be more sensible to add an error displayed on the UI that says `Invalid number entered` but that capability is not embedded here.